build-tools/build.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+git config --global --add safe.directory /github/workspace
+
+# Create a github group
+echo "::group::Debug Variables"
+
+# -------------------
+# DEBUGGING VARIABLES
+# -------------------
+
+# Echo all variables starting with GITHUB_ for debugging
+for var in "${!GITHUB_@}"; do
+    echo "$var=${!var}"
+done
+
+# Echo all variables starting with INPUT_ for debugging
+for var in "${!INPUT_@}"; do
+    echo "$var=${!var}"
+done
+
+# echo all variables starting with RUNNER_ for debugging
+for var in "${!RUNNER_@}"; do
+    echo "$var=${!var}"
+done
+
+
+echo "::endgroup::"
+# ----------------
+# CHECK IF TEMPLATE
+# ----------------
+echo "::group::Checking if this is a template"
+
+template=$(awk -F'=' '/^IS_LIBRARY:=/{print $2}' Makefile)
+if [ "$template" == "1" ]; then
+    echo "is template"
+else
+    echo "is not template"
+fi
+echo "template=$template" >> $GITHUB_OUTPUT
+
+echo "::endgroup::"
+# ----------------
+# GET PROJECT INFO
+# ----------------
+echo "::group::Getting project info"
+
+if [ "$INPUT_ACTION" == "opened" ]; then
+
+    # Fetch the head SHA directly from the PR API
+    API_URL="https://api.github.com/repos/$GITHUB_REPOSITORY/pulls/$GITHUB_PR_NUM"
+    echo "API URL: $API_URL"
+    API_RESPONSE=$(wget -O- --quiet "$API_URL")
+    if [ $? -ne 0 ]; then
+        echo "Error fetching data from GitHub API"
+        exit $?
+    fi
+    echo "API Response: $API_RESPONSE"
+    sha=$(echo "$API_RESPONSE" | jq -r '.head.sha' | head -c 6)
+
+else
+    # Use the commit SHA after the event
+    sha=$(git rev-parse HEAD | head -c 6)
+fi
```

Comment:
It's possible this entire section could be reduced to 
```sh
sha=$(echo $GITHUB_SHA | head -c 6)
```

I'd need to test with the PR side though to check if the sha changes with new commits.